src/infobar.rs

```diff
@@ -3,32 +3,52 @@
 //! Module InfoBar provides structures and functions for building and rendering an infobar
 
 use crate::paths::Paths;
+use crate::ui::Mode;
 
 /// Text contains the strings required to print the infobar.
 pub struct Text {
-    /// current image is the string of the current image path
-    pub current_image: String,
-    /// index is the string represention of the index.
-    pub index: String,
+    /// Either displays the name of the current image or the current command the user is typing in
+    /// command mode
+    pub information: String,
+    /// In normal mode this is the string represention of the index, in command mode this is
+    /// "Command"
+    pub mode: String,
 }
 
-impl From<&Paths> for Text {
-    fn from(p: &Paths) -> Self {
-        let current_image = match p.images.get(p.index) {
-            Some(path) => match path.to_str() {
-                Some(name) => name.to_string(),
-                None => "No file".to_string(),
-            },
-            None => "No file selected".to_string(),
+impl Text {
+    /// Updates the infobar based on the current mode of the applicaiton
+    /// Normal Mode:
+    ///     mode = index of current image
+    ///     information = path to current image
+    /// Command Mode:
+    ///     mode = "Command"
+    ///     information = curerntly entered user string
+    /// Error Mode:
+    ///     mode = "Error"
+    ///     information = error message to display
+    pub fn update(current_mode: &Mode, paths: &Paths) -> Self {
+        let (mode, information) = match current_mode {
+            Mode::Command(msg) => ("Command".to_string(), format!(":{}", msg)),
+            Mode::Normal => {
+                let information: String;
+                let mode: String;
+                information = match paths.images.get(paths.index) {
+                    Some(path) => match path.to_str() {
+                        Some(name) => name.to_string(),
+                        None => "No file".to_string(),
+                    },
+                    None => "No file selected".to_string(),
+                };
+                mode = if paths.images.is_empty() {
+                    "No files in path".to_string()
+                } else {
+                    format!("{} of {}", paths.index + 1, paths.max_viewable)
+                };
+                (mode, information)
+            }
+            Mode::Error(msg) => ("Error".to_string(), msg.to_string()),
+            _ => (String::new(), String::new()),
```

Comment:
@NickHackman 
(50) This is the line.

The program renders the infobar for the `Exit` mode which returns an empty string for both `mode` and `information`